theano/tensor/nnet/abstract_conv.py

```diff
@@ -1565,11 +1572,11 @@ def conv(self, img, kern, mode="valid", dilation=1, unshared=False):
                             if unshared is True:
                                 for row in xrange(kern.shape[2]):
                                     for col in xrange(kern.shape[3]):
-                                        val = _valfrommode('valid')
-                                        out[b, n, row, col] += _convolve2d(img[b, im0, row:row + kern.shape[4],
-                                                                           col:col + kern.shape[5]],
-                                                                           dilated_kern[n, im0, row, col, ...],
-                                                                           1, val, bval, 0)
+                                        for k_row in xrange(kern.shape[4]):
+                                            for k_col in xrange(kern.shape[5]):
+                                                out[b, n, row, col] += img[b, im0, row + k_row,
+                                                                           col + k_col] * \
+                                                    dilated_kern[n, im0, row, col, k_row, k_col]
```

Comment:
I was actually thinking of doing an `np.sum(np.multiply(..., ...))` to eliminate the two inner loops and not unnecessarily call `_convolve2d`. Do you think I should?